### PR TITLE
Makes branding migration synchronous and re-runnable

### PR DESCRIPTION
**Pull request by SCdF**
Branding migration now passes promises back correctly, meaning the
migration controller correctly waits until it's done before running more
migrations.

The migration also doesn't conflict if the branding document already
exists, which may occur if you're re-running migrations for some reason.

### CODE REVIEW

**Comment by SCdF** [comment]
@garethbowen / @Jodge adding you both as I'm not sure if I'm stomping over anything intentional with some of my choices.

At the very least we should change the migration to be re-runnable: it's important for development and fixing problems in the field (occasionally), and is IMO generally a good plan to follow.

File: api/src/migrations/add-branding-doc.js [hunk 1]
```diff
@@ -32,42 +33,45 @@ const attachDocument = (src, resource) => {
...
-      const doc = {
-        _id: BRANDING_ID,
-        title: appTitle,
-        resources: {
-          [logo.name]: logo.file,
-          [favicon.name]: favicon.file
+  return Promise.all([
+    attachDocument(srcLogo, logo),
+    attachDocument(srcFav, favicon)
+  ]).then(attachments => {
```
Comment (garethbowen) [inline api/src/migrations/add-branding-doc.js]: Can we use destructuring to name the attachments? eg:

```js
   ]).then(([ logo, favicon ]) => {
```

**Comment by garethbowen** [comment]
@Jodge What do you think of this change? Does it make sense?

**Comment by SCdF** [comment]
@Jodge sorry if there was a misunderstanding! The intention was that you would review this PR. It is my job as the contributor to contribute to it :-)

All developers at Medic review each others code. Here I specifically want you to review this because you contributed it originally, and I want to make sure I've understood the code when I've made my changes, and I want to also make sure you understand my changes.

Let me know over Slack if you're not sure how to approach reviewing :-)

**Pull request merged**